File: frontend_flutter/lib/modules/production/domain/entities/production_plan.dart
/// Domain entity. No Flutter, no JSON.
/// Bảng Kế hoạch Sản xuất: ngày lập, danh sách SP (MÃ SP, TÊN SP, QUY CÁCH, DẠNG, DẠNG ĐÓNG GÓI, SỐ).
class ProductionPlanItem {
  const ProductionPlanItem({
    required this.ordinal,
    required this.productCode,
    required this.productName,
    required this.specification,
    required this.productForm,
    required this.packagingForm,
    required this.plannedQuantity,
  });

  final int ordinal;
  final String productCode;
  final String productName;
  final String specification;
  final String productForm;
  final String packagingForm;
  final int plannedQuantity;
}

/// Plan status from API: draft | submitted | approved.
class ProductionPlan {
  const ProductionPlan({
    required this.planDate,
    required this.items,
    this.id,
    this.status,
  });

  final String planDate;
  final List<ProductionPlanItem> items;
  /// Set when loaded from API; null when no plan for date.
  final String? id;
  /// draft | submitted | approved
  final String? status;
}

File: backend/internal/dto/production_plan.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

// ProductionPlanItemPayload for request (create/update).
type ProductionPlanItemPayload struct {
	ProductID       uuid.UUID `json:"productId"`
	Ordinal         int       `json:"ordinal"`
	PlannedQuantity int       `json:"plannedQuantity"`
}

// CreateProductionPlanRequest for POST production-plans.
type CreateProductionPlanRequest struct {
	PlanDate string                     `json:"planDate"` // YYYY-MM-DD
	Items    []ProductionPlanItemPayload `json:"items"`
}

// UpdateProductionPlanRequest for PUT production-plans/:id.
type UpdateProductionPlanRequest struct {
	PlanDate string                     `json:"planDate"`
	Items    []ProductionPlanItemPayload `json:"items"`
}

// RejectProductionPlanRequest for POST production-plans/:id/reject.
type RejectProductionPlanRequest struct {
	Reason string `json:"reason"`
}

// ProductionPlanItemDetail for response (includes product display info: MÃ SP, TÊN, QUY, DẠNG, ĐÓNG GÓI).
type ProductionPlanItemDetail struct {
	ID              uuid.UUID `json:"id"`
	ProductID       uuid.UUID `json:"productId"`
	ProductCode     string    `json:"productCode"`
	ProductName     string    `json:"productName"`
	Specification   string    `json:"specification"`   // PackageSize + PackageUnit (e.g. 100gr)
	ProductForm     string    `json:"productForm"`     // ProductType: Bột uống, Dung dịch, ...
	PackagingForm   string    `json:"packagingForm"`   // PackagingType: Gói, Chai, ...
	Ordinal         int       `json:"ordinal"`
	PlannedQuantity int       `json:"plannedQuantity"`
}

// ProductionPlanPayload for list/get.
type ProductionPlanPayload struct {
	ID         uuid.UUID                  `json:"id"`
	PlanDate   time.Time                  `json:"planDate"`
	Status     string                     `json:"status"`
	CreatedAt  time.Time                  `json:"createdAt"`
	CreatedBy  uuid.UUID                  `json:"createdBy"`
	ApprovedAt *time.Time                  `json:"approvedAt,omitempty"`
	ApprovedBy *uuid.UUID                 `json:"approvedBy,omitempty"`
	Items      []ProductionPlanItemDetail `json:"items"`
}
